addons/sale/models/sale_order.py

```diff
@@ -1690,6 +1776,84 @@ def _get_default_payment_link_values(self):
             'amount_paid': self.amount_paid,
         }
 
+    # EDI #
+
+    def create_document_from_attachment(self, attachment_ids):
+        """ Create the sale orders from given attachment_ids and redirect newly create order view.
+
+        :param list attachment_ids: List of attachments process.
+        :return: An action redirecting to related sale order view.
+        :rtype: dict
+        """
+        orders = self._create_order_from_attachment(attachment_ids)
+        return orders._get_records_action(name=_("Generated Orders"))
+
+    @api.model
+    def _create_order_from_attachment(self, attachment_ids):
+        """ Create the sale orders from given attachment_ids and fill data by extracting detail
+        from attachments and return generated orders.
+
+        :param list attachment_ids: List of attachments process.
+        :return: Recordset of order.
+        """
+        attachments = self.env['ir.attachment'].browse(attachment_ids)
+        if not attachments:
+            raise UserError(_("No attachment was provided"))
+
+        orders = self.browse()
+        for attachment in attachments:
+            order = self.create({
+                'partner_id': self.env.user.partner_id.id,
+            })
+            order._extend_with_attachments(attachment)
+            orders |= order
+            order.message_post(attachment_ids=attachment.ids)
```

Comment:
Why do you post a message here? 
Is it only to link the attachments to the chatter ? :thinking:
It seems strange to call `message_post` without any message

Reply:
Because adding file in chatter we are extracting data from
I guess we can add some message like `Extracted order from file` or something like that :thinking: 
Same thing is done in lot of other places only post attachments without message (e.g. in invoice)

Reply:
AFAIK, simply linking the file to the record (with res_id & res_model) is sufficient and there shouldn't be a need to call `message_post`.  We can live with that for now, but I believe it could be avoided in the future.